channels/consumer.py

```diff
@@ -47,6 +47,10 @@ async def __call__(self, scope, receive, send):
             self.channel_receive = functools.partial(
                 self.channel_layer.receive, self.channel_name
             )
+            if getattr(self.channel_layer, "clean_channel", None) and callable(self.channel_layer.clean_channel):
```

Comment:
Future refactoring might be easier if you try/except an AttributeError here (if `self.channel_layer.clean_channel` doesn't exist, it will throw). This removes the hardcoded string which can make refactoring, type checking, etc fall over.

Reply:
good call, you mean something like

```python3
try:
  if callable(self.channel_layer.clean_channel):
    cancel_callback = functools.partial(self.channel_layer.clean_channel, self.channel_name)
except AttributeError:
  cancel_callback = None
```

?

Reply:
Yes, exactly what I had in mind

Reply:
thanks, changed it